src/Sarif.Multitool/Rules/SARIF2001.AuthorHighQualityMessages.cs

```diff
@@ -3,85 +3,106 @@
 
 using System;
 using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
 using Microsoft.Json.Pointer;
 
 namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
 {
     public class AuthorHighQualityMessages : SarifValidationSkimmerBase
     {
-        public override MultiformatMessageString FullDescription => new MultiformatMessageString
-        {
-            Text = RuleResources.SARIF2001_AuthorHighQualityMessages_FullDescription_Text
+        /// <summary>
+        /// SARIF2001
+        /// </summary>
+        public override string Id => RuleId.AuthorHighQualityMessages;
+
+        /// <summary>
+        /// Placeholder (full description).
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2001_AuthorHighQualityMessages_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_EnquoteDynamicContent_Text),
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_TerminateWithPeriod_Text)
         };
 
         public override FailureLevel DefaultLevel => FailureLevel.Warning;
 
-        public override string Id => RuleId.AuthorHighQualityMessages;
+        private static readonly Regex s_dynamicContentRegex = new Regex(@"{[0-9]+}", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static readonly Regex s_enquoteDynamicContentRegex = new Regex(@"'{[0-9]+}'", RegexOptions.Compiled | RegexOptions.CultureInvariant);
 
-        protected override IEnumerable<string> MessageResourceNames
+        protected override void Analyze(Tool tool, string toolPointer)
         {
-            get
+            if (tool.Driver != null)
             {
-                return new string[]
-                {
-                    nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_TerminateWithPeriod_Text)
-                };
+                AnalyzeToolDriver(tool.Driver, toolPointer.AtProperty(SarifPropertyName.Driver));
             }
         }
 
-        protected override void Analyze(ReportingDescriptor reportingDescriptor, string reportingDescriptorPointer)
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolDriverPointer)
         {
-            AnalyzeMessageStrings(reportingDescriptor.MessageStrings, reportingDescriptorPointer);
+            if (toolComponent.Rules != null && toolComponent.Rules.Count > 0)
```

Comment:
> && toolComponent.Rules.Count > 0 [](start = 43, length = 33)

You don't need this check because if Rules is empty, then the `foreach` loop will simply execute 0 times.